In [22]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [23]:
data = pd.read_csv("Clean_DDoS.csv")

In [32]:
data.head()

Source IP   Source Port   Destination IP   Destination Port   Protocol  \
0           0           443                1              54865          0   
1           0            80                1              55054          0   
2           0            80                1              55055          0   
3           0           443                1              46236          0   
4           0           443                1              54863          0   

   Total Length of Fwd Packets   Fwd Packet Length Max  \
0                           12                       6   
1                            6                       6   
2                            6                       6   
3                            6                       6   
4                           12                       6   

    Fwd Packet Length Min   Fwd Packet Length Mean   Fwd Packet Length Std  \
0                       6                      6.0                     0.0   
1                       6                      6.0                     0.0   
2                       6                      6.0                     0.0   
3                       6                      6.0                     0.0   
4                       6                      6.0                     0.0   

   ...   SYN Flag Count   PSH Flag Count   URG Flag Count   Down/Up Ratio  \
0  ...                0                0                0               0   
1  ...                0                0                1               1   
2  ...                0                0                1               1   
3  ...                0                0                1               1   
4  ...                0                0                0               0   

    Average Packet Size   Avg Fwd Segment Size   Avg Bwd Segment Size  \
0                   9.0                    6.0                    0.0   
1                   9.0                    6.0                    6.0   
2                   9.0                    6.0                    6.0   
3                   9.0                    6.0                    6.0   
4                   9.0                    6.0                    0.0   

    Subflow Fwd Bytes   min_seg_size_forward   Label  
0                  12                     20     0.0  
1                   6                     20     0.0  
2                   6                     20     0.0  
3                   6                     20     0.0  
4                  12                     20     0.0  

[5 rows x 34 columns]

In [25]:
print(data[" Source Port"].value_counts().count(), "Source port")
print(data[" Destination Port"].value_counts().count(), "Destination port")

50697 Source port
23950 Destination port


In [26]:
data[" Protocol"].value_counts()

 Protocol
6     192820
17     32871
0         54
Name: count, dtype: int64

In [27]:
data = data[ data[" Protocol"] != 0]

In [28]:
from sklearn.preprocessing import LabelEncoder

In [30]:
encoder = LabelEncoder()
data[" Protocol"] = encoder.fit_transform(data[[" Protocol"]])

c:\Users\acasalro\AppData\Local\miniconda3\envs\ia\Lib\site-packages\sklearn\preprocessing\_label.py:114: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [31]:
# 0 = 6
# 1 = 17
data[" Protocol"].value_counts()

 Protocol
0    192820
1     32871
Name: count, dtype: int64

SOURCE

80 - 443 = Web
137 - 138 - 139 = Netbios
22 = SSH
21 = FTP
389 = LDAP
123 = NTP

Destination

80 - 443 - 8080 = Web
53 = UDP
88 = Kerberos
465 = SMTP
137 - 138 - 139 - 445 = Netbios
22 = SSH
21 = FTP
389 - 3268 = LDAP
135 = RPC
123 = NTP
5353 = DNS

In [44]:
vc = data[" Destination Port"].value_counts()
vc[vc > 10]

 Destination Port
80       136951
53        31950
443       13485
8080        510
123         362
22          342
137         274
389         261
88          173
21          167
465         147
139         100
3268         91
138          53
445          48
49666        23
135          23
5353         16
49671        12
Name: count, dtype: int64